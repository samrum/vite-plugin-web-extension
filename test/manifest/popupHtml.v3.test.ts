import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("popupHtml");

runManifestV3Test("popupHtml", () => ({
  action: {
    default_popup: `${resourceDir}/popup.html`,
  },
}));
