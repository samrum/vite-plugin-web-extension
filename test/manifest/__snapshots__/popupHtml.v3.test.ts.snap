// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`popupHtml - Manifest V3 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
log("popup");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/popupHtml/popup.html",
    "fileName": "assets/test/manifest/resources/popupHtml/popup.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/popupHtml/popup.html": {
        "code": "",
        "originalLength": 224,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/popupHtml/popup.js": {
        "code": "log("popup");",
        "originalLength": 50,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 13,
      },
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "test/manifest/resources/popupHtml/popup",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  "version": "1.0.0",
  "name": "Manifest Name",
  "manifest_version": 3,
  "action": {
    "default_popup": "test/manifest/resources/popupHtml/popup.html"
  }
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/popupHtml/popup.html",
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />

    <script type="module" src="http://example.com/httpModuleScript.js"></script>
    <script type="module" crossorigin src="/assets/test/manifest/resources/popupHtml/popup.js"></script>
  </head>
</html>
",
    "type": "asset",
  },
]
`;
